share/completions/npm.fish

```diff
@@ -70,22 +70,14 @@ end
 # and: https://github.com/fish-shell/fish-shell/pull/2366
 complete -f -c npm -n 'not __fish_npm_needs_option; and not __fish_npm_using_command run; and not __fish_npm_using_command run-script' -a "(__fish_complete_npm)"
 
-# list available npm scripts and their parial content
-function __fish_parse_npm_run_completions
-    while read -l name
-        set -l trim 20
-        read -l value
-        set value (string sub -l $trim -- $value)
-        printf "%s\t%s\n" $name $value
-    end
-end
-
 function __fish_npm_run
     # Like above, only try to call npm if there's a command by that name to facilitate aliases that call nvm.
     if command -sq jq; and test -e package.json
-        jq -r '.scripts | to_entries[] | .key,.value' <package.json | __fish_parse_npm_run_completions
+        jq -r '.scripts | to_entries | map("\(.key)\t\(.value | tostring | .[0:20])") | .[]' package.json
+    else if command -sq npm; and command -sq jq
+        command npm run --json | jq -r 'to_entries | map("\(.key)\t\(.value | tostring | .[0:20])") | .[]'
     else if command -sq npm
-        command npm run | string match -r -v '^[^ ]|^$' | string trim | __fish_parse_npm_run_completions
+        command npm run | string match -r -v '^[^ ]|^$' | string trim | awk 'NR % 2 == 1 { name=$0 ; next } { print name "\t" substr($0, 0, 20) }'
```

Comment:
The function has the advantage that it uses only builtins. With awk, it's a bit of a crapshoot which things are supported on which implementation, so we try not to rely on it unless it can't be helped.

Is there a specific reason to change this other than that it's shorter?

Reply:
Thats a good point about different awk implementations having different behaviors. I only tested this on my mac, (`GNU Awk 4.2.0, API: 2.0 (GNU MPFR 3.1.6, GNU MP 6.1.2)
`), so I don't really know if this will break elsewhere.

There was no specific reason other than it's shorter and avoided defining another function. I can revert the changes to this file if you want.

Reply:
>I can revert the changes to this file if you want.

No need, it's in a different commit already. I can just not merge that one.